src/windows/main_window.py

```diff
@@ -2232,25 +2232,65 @@ def actionInsertTimestamp_trigger(self, event):
         app = get_app()
         _ = app._tr
 
+        if len(self.selected_effects) == 0:
+            log.info("No caption effect selected")
+            return
+        effect_data = Effect.filter(id=self.selected_effects[0])[0].data
+        effect_id = effect_data.get("id")
+        if effect_data.get("type") != "Caption":
+            log.info("Captioning an effect that is not a Caption")
+            return
+
+        # Get the Clip that owns this caption effect
+        clip_data = None
+        for clip in Clip.filter():
```

Comment:
Clip.filter() will return all clips in the project. It would probably be better if you could get the Clip ID of the parent clip, and pass it to the filter() function. Not sure if we have the Clip ID in the effect data though.

Reply:
I just checked `clip_data.get("id") in str(effect_data)` -> `false`. Not seeing a way to get clip id out of the effect